libpod/container_internal_linux.go

```diff
@@ -88,6 +88,10 @@ func (c *Container) prepare() (Err error) {
 				c.state.NetworkStatus = networkStatus
 			}
 		}
+
+		if c.config.NetMode == "slirp4netns" && !c.config.PostConfigureNetNS {
+			rootlessSetupErr = c.runtime.setupRootlessNetNS(c)
```

Comment:
Error handling flow is broken for this one - if this fails we need to report it much earlier than that `return` below to ensure proper cleanup of resources on error. I think it'd be better to reuse `createNetNSErr` here - it already has the right logic here.